dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentAsDraftActionlet.java

```diff
@@ -0,0 +1,123 @@
+package com.dotmarketing.portlets.workflows.actionlet;
+
+import com.dotmarketing.beans.Permission;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.RelationshipAPI;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.categories.business.CategoryAPI;
+import com.dotmarketing.portlets.categories.model.Category;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.structure.model.Relationship;
+import com.dotmarketing.portlets.workflows.model.*;
+import com.dotmarketing.util.Logger;
+import com.liferay.portal.model.User;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Do the save as draft for a content (checkin)
+ * @author jsanca
+ */
+public class SaveContentAsDraftActionlet extends WorkFlowActionlet {
+
+	private final ContentletAPI   contentletAPI;
+	private final RelationshipAPI relationshipAPI;
+	private final CategoryAPI     categoryAPI;
+	private final PermissionAPI   permissionAPI;
+
+	public SaveContentAsDraftActionlet() {
+
+		this.contentletAPI   = APILocator.getContentletAPI();
+		this.relationshipAPI = APILocator.getRelationshipAPI();
+		this.categoryAPI     = APILocator.getCategoryAPI();
+		this.permissionAPI   = APILocator.getPermissionAPI();
+	}
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 1L;
+
+	public String getName() {
+		return "Save Draft content";
+	}
+
+	public String getHowTo() {
+
+		return "This actionlet will do checkin as draft the content.";
+	}
+
+	public void executeAction(final WorkflowProcessor processor,
+							  final Map<String, WorkflowActionClassParameter> params) throws WorkflowActionFailureException {
+
+		final Map<Relationship, List<Contentlet>> contentRelationships = new HashMap<Relationship, List<Contentlet>>();
+
+		try {
+
+			final Contentlet contentlet             = processor.getContentlet();
+			final User user                         = processor.getUser();
+			final List<Category > categories        =
+					this.categoryAPI.getParents(contentlet, user, false);
+			final List< Permission > permissions    =
+					this.permissionAPI.getPermissions(contentlet, false, true);
+
+			Logger.debug(this,
+					"Saving the content as draft of the contentlet: " + contentlet.getIdentifier());
+
+			contentlet.setProperty(Contentlet.WORKFLOW_IN_PROGRESS, Boolean.TRUE);
+
+			this.buildRelationships(processor.getUser(), contentRelationships, contentlet);
+
+			this.contentletAPI.saveDraft(
+					processor.getContentlet(), contentRelationships, categories, permissions, user, true);
+
+			Logger.debug(this,
+					"content draft already saved for the contentlet: " + contentlet.getIdentifier());
+		} catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)